snetd/cmd/components.go

```diff
@@ -394,6 +405,11 @@ type VerifyMeteringResponse struct {
 
 func (components *Components) GrpcPaymentValidationInterceptor() grpc.StreamServerInterceptor {
 	if !components.Blockchain().Enabled() {
+		if config.GetBool(config.AllowedUserFlag) {
+			log.Info("Blockchain is disabled And AllowedUserFlag is enabled")
+			return handler.AllowedUserValidationInterceptor(components.AllowedUserPaymentHandler())
```

Comment:
After implementing separate payment handler additional interceptor is not needed anymore:
```suggestion
			return handler.GrpcPaymentValidationInterceptor(components.AllowedUserPaymentHandler())
```
You can delete `allowedUserValidationInterceptor` and all related functions.